docs/test_cases/TestSuite42.md

```diff
@@ -0,0 +1,205 @@
+
+From View/Edit BIE page, select the desired BIE, select the desired node under the BIE, the end user can click the button "Assign New Business Term" to assign new business terms to that node. 
+
+
+### Test Step Pre-condition:
+1. There are at least three BIEs created by end user and developer and they are in different states (WIP, QA, Production). 
+2. There are at least five business terms created. Three of them are assigned to different BIEs. At least Two of them are not assigned to any BIEs. 
+
+
+### Test Step:
+1. An end user logins.
+2. He visits the Home page and clicks on the BIE tab.
+3. He clicks on the "View/Edit BIE" from the dropdown list.
+4. He clicks on any BIE from the table. The BIE detail page will open.
+5. He expands the root level in the left sidenav tree and clicks on any used node in the second level, the detail for that node will disply on the right.
+6. He will click on "Show Business Terms" button to view all the business terms that already assigned to that node.
+7. Verify that the list of business terms in the new window is indeed assigned to the selected node (Assertion [#1](#test-assertion-4231)).
+8. He closes the new window and goes back to previous window.
+9. He will hover over "Show Business Terms" button without clicking. A hint bar will display to show the number of business terms assigned to the node. Verify the number is the same as in the previosu step. (Assertion [#2](#test-assertion-4232)).
+10. He will clicks on the "Assign New Business Term" button right next to "Show Business Term" button on the same page.
+11. He will select any availabe business term in the new window,  set the Type Code, check or uncheck the Preferred Business term. Then clicks on "Create".
+12. He will close the new window and go back to previous window. 
+13. He will hover over "Show Business Terms" button without clicking again. Verify the number in the hint bar is increased by one. (Assertion [#3](#test-asssertion-4233)).
+
+
+## Test Case 42.4
+
+**Business Term from External Source**
+
+Pre-condition: N/A
+
+### Test Assertion:
+
+#### Test Assertion #42.4.1
+Under BIE, find View/Edit Business Term,  the end user can click "Upload Business Terms", and then they can click the "Download template" from the "Upload Business Terms from File" page.
+
+#### Test Assertion #42.4.2
+
+Under BIE, find View/Edit Business Term,  the end user can click "Upload Business Terms", and then they can click on the Attach button and choose the file from the computer, and the file should be uploaded. 
+
+#### Test Assertion #42.4.3
+
+For bulk upload through "Upload Business Terms", if the business term is uploaded with new externalReferenceUri, a new business term will be created. 
+
+#### Test Assertion #42.4.4
+
+For bulk upload through "Upload Business Terms", if the business term is uploaded with an existent externalReferenceUri, the previous business term with the same URI will be updated with the new information. 
+
```

Comment:
@bingqi2045 You might want to add some TAs addressing situations when certain information is missing for some particular rows in the uploaded file, i.e., cases where bad data are supplied, e.g., missing columns, missing header row, too many column, invalid data format.